src/_findErroneousUnionNode.ts

```diff
@@ -0,0 +1,78 @@
+import { Cube, LevelCoordinates } from "@activeviam/data-model";
+import {
+  findDescendant,
+  isMdxFunction,
+  Mdx,
+  MdxFunction,
+} from "@activeviam/mdx";
+
+import { _doesCrossjoinRepresentAnExpandedMember } from "./_doesCrossjoinRepresentAnExpandedMember";
+import { _doesCrossjoinYieldAllCombinationsOfMembers } from "./_doesCrossjoinYieldAllCombinationsOfMembers";
+
+/**
+ * The shape of an erroneous union node.
+ * Erroneous union nodes can lead to errors and must be cleaned up.
+ */
+export interface ErroneousUnionNode extends MdxFunction {
+  name: "Union";
+  arguments: (MdxFunction & {
+    name: "Crossjoin";
+  })[];
+}
+
+/**
+ * Returns whether `mdx` represents an erroneous union node.
+ */
+export function _findErroneousUnionNode(
+  mdx: Mdx,
+  { cube, levelsOnAxis }: { cube: Cube; levelsOnAxis: LevelCoordinates[] }
+): { path: (string | number)[]; match: ErroneousUnionNode } | undefined {
+  const result = findDescendant(mdx, (node) => {
+    if (!isMdxFunction(node, "union") || node.arguments.length < 2) {
+      // The erroneous node is an union function, with at least 2 arguments.
```

Comment:
Reading this comment inside the if statements feels misleading, the if statement represent the negation of the comment ! I would negate the sentence or move it before the if statement with a `should form`
`// The erroneous node is not a union function, with at least 2 arguments.`
or
`// The erroneous node should be an union function, with at least 2 arguments.`

Reply:
The first proposal does not work, as it conveys the opposite of reality here. The second could work, but I don't like it as much.
The reality here is that: _"erroneous node = node to update = union function whose arguments are all crossjoins."_. Maybe I could rewrite the comment that way.

About the _"should form"_, I always avoid it. My reason is that it creates confusion by nature (it being conditional), just like _"might"_. For instance, if you are debugging some code with a comment that say:
```
// `a` is a number.
```
and you realize that during runtime, `a` is in fact not a number, then you know that there **is** a problem with `a`.
Whereas if the comment had said:
```
// `a` should be a number.
```
then you will think that there **might be** a problem with `a`.

And I want you to know that there is a problem, not that there might be one.

In short, I believe that the most direct / unambiguous way of writing is always better, and that the more confused way is always worst.